File: sportsbook-app/payment-validator-service/src/main/java/com/sportsbook/payment/validator/domain/DepositDecisionEntity.java
package com.sportsbook.payment.validator.domain;

import com.sportsbook.payment.contracts.DepositDecision;
import com.sportsbook.payment.contracts.DepositRejectionReason;
import com.sportsbook.payment.contracts.FundingSourceType;
import jakarta.persistence.*;

import java.math.BigDecimal;
import java.time.Instant;

@Entity
@Table(name = "deposit_decision")
public class DepositDecisionEntity {

    @Id
    private String depositId;

    private String playerId;

    private BigDecimal amount;

    @Enumerated(EnumType.STRING)
    private FundingSourceType fundingSourceType;

    @Enumerated(EnumType.STRING)
    private DepositDecision decision;

    @Enumerated(EnumType.STRING)
    private DepositRejectionReason rejectionReason;

    private String providerReference;

    private Instant decidedAt;

    public String getDepositId() { return depositId; }
    public void setDepositId(String depositId) { this.depositId = depositId; }
    public String getPlayerId() { return playerId; }
    public void setPlayerId(String playerId) { this.playerId = playerId; }
    public BigDecimal getAmount() { return amount; }
    public void setAmount(BigDecimal amount) { this.amount = amount; }
    public FundingSourceType getFundingSourceType() { return fundingSourceType; }
    public void setFundingSourceType(FundingSourceType fundingSourceType) { this.fundingSourceType = fundingSourceType; }
    public DepositDecision getDecision() { return decision; }
    public void setDecision(DepositDecision decision) { this.decision = decision; }
    public DepositRejectionReason getRejectionReason() { return rejectionReason; }
    public void setRejectionReason(DepositRejectionReason rejectionReason) { this.rejectionReason = rejectionReason; }
    public String getProviderReference() { return providerReference; }
    public void setProviderReference(String providerReference) { this.providerReference = providerReference; }
    public Instant getDecidedAt() { return decidedAt; }
    public void setDecidedAt(Instant decidedAt) { this.decidedAt = decidedAt; }
}

File: sportsbook-app/payment-validator-service/src/main/kotlin/com/sportsbook/payment/validator/domain/DepositDecisionEntity.kt
package com.sportsbook.payment.validator.domain

import com.sportsbook.payment.contracts.DepositDecision
import com.sportsbook.payment.contracts.DepositRejectionReason
import com.sportsbook.payment.contracts.FundingSourceType
import jakarta.persistence.Entity
import jakarta.persistence.EnumType
import jakarta.persistence.Enumerated
import jakarta.persistence.Id
import jakarta.persistence.Table
import java.math.BigDecimal
import java.time.Instant

@Entity
@Table(name = "deposit_decision")
class DepositDecisionEntity {

    @Id
    var depositId: String? = null

    var playerId: String? = null

    var amount: BigDecimal? = null

    @Enumerated(EnumType.STRING)
    var fundingSourceType: FundingSourceType? = null

    @Enumerated(EnumType.STRING)
    var decision: DepositDecision? = null

    @Enumerated(EnumType.STRING)
    var rejectionReason: DepositRejectionReason? = null

    var providerReference: String? = null

    var decidedAt: Instant? = null
}
